worlds/yugioh06/__init__.py

```diff
@@ -0,0 +1,454 @@
+            normal = [e for e in challenge if e not in excluded and e not in prio]
+            total = list(excluded) + normal + list(prio)
+            self.removed_challenges = total[:noc]
+
+        self.campaign_opponents = get_opponents(self.multiworld, self.player,
+                                                self.options.campaign_opponents_shuffle.value)
+
+    def fill_slot_data(self) -> Dict[str, Any]:
+        slot_data: Dict[str, Any] = {
+            "structure_deck": self.options.structure_deck.value,
+            "banlist": self.options.banlist.value,
+            "final_campaign_boss_unlock_condition": self.options.final_campaign_boss_unlock_condition.value,
+            "fourth_tier_5_campaign_boss_unlock_condition": self.options.fourth_tier_5_campaign_boss_unlock_condition.value,
+            "third_tier_5_campaign_boss_unlock_condition": self.options.third_tier_5_campaign_boss_unlock_condition.value,
+            "final_campaign_boss_challenges": self.options.final_campaign_boss_challenges.value,
+            "fourth_tier_5_campaign_boss_challenges": self.options.fourth_tier_5_campaign_boss_challenges.value,
+            "third_tier_5_campaign_boss_challenges": self.options.third_tier_5_campaign_boss_campaign_opponents.value,
+            "final_campaign_boss_campaign_opponents": self.options.final_campaign_boss_campaign_opponents.value,
+            "fourth_tier_5_campaign_boss_campaign_opponents": self.options.fourth_tier_5_campaign_boss_unlock_condition.value,
+            "third_tier_5_campaign_boss_campaign_opponents": self.options.third_tier_5_campaign_boss_campaign_opponents.value,
+            "number_of_challenges": self.options.number_of_challenges.value,
+        }
+
+        slot_data["removed challenges"] = self.removed_challenges
+        slot_data["starting_booster"] = self.starting_booster
+        slot_data["starting_opponent"] = self.starting_opponent
+        return slot_data
+
+    # for the universal tracker, doesn't get called in standard gen
+    @staticmethod
+    def interpret_slot_data(slot_data: Dict[str, Any]) -> Dict[str, Any]:
+        # returning slot_data so it regens, giving it back in multiworld.re_gen_passthrough
+        return slot_data
+
+    def generate_output(self, output_directory: str):
+        #patched_rom = self.apply_randomizer()
+        outfilebase = 'AP_' + self.multiworld.seed_name
+        outfilepname = f'_P{self.player}'
+        outfilepname += f"_{self.multiworld.get_file_safe_player_name(self.player).replace(' ', '_')}"
+        outputFilename = os.path.join(output_directory, f'{outfilebase}{outfilepname}.gba')
+        self.rom_name_text = f'YGO06{Utils.__version__.replace(".", "")[0:3]}_{self.player}_{self.multiworld.seed:11}\0'
+        self.romName = bytearray(self.rom_name_text, 'utf8')[:0x20]
+        self.romName.extend([0] * (0x20 - len(self.romName)))
+        self.rom_name = self.romName
```

Comment:
But why? What is the difference between `romName` and `rom_name`, and why is one assigned to the other by the end anyway?